definitions/checks/katello_agent.rb

```diff
@@ -0,0 +1,38 @@
+require 'yaml'
+
+class Checks::CheckKatelloAgentEnabled < ForemanMaintain::Check
+  metadata do
+    label :check_katello_agent_enabled
+    description 'Check whether the katello-agent feature is enabled before upgrading'
+    tags :pre_upgrade
+  end
+
+  confine do
+    feature(:instance).downstream
+  end
+
+  INSTALLER_ANSWERS_FILE = '/etc/foreman-installer/scenarios.d/satellite-answers.yaml'.freeze
+
+  def run
+    assert(
+      !katello_agent_enabled?,
+      "The katello-agent feature is enabled on this system. As of Satellite 6.15.0, katello-agent"\
+      " is removed and will no longer function. Before proceeding with the upgrade, you should ensure"\
+      " that you have deployed and configured an alternative tool for remote package management and"\
+      " patching for content hosts, such as the Remote Execution (REX) Pull Provider. See the"\
```

Comment:
Since technically it's not a "transport" any more, they standardized on different terms in the [documentation](https://docs.theforeman.org/nightly/Installing_Proxy/index-katello.html#enabling-remote-execution_smart-proxy):

```suggestion
      " patching for content hosts, such as remote execution (REX) with pull-based transport. See the"\
```